### PR TITLE
[PREVIEW] [ISPN-7782] Generate documentation code snippets from testsuite

### PR DESCRIPTION
**Pull request by Holmistr**
This is a preview, which should server as a subject to debate. I will send an email to dev mailing list soon opening the discussion.

### CODE REVIEW

File: core/src/test/java/org/infinispan/configuration/ConfigurationUnitTest.java [hunk 1]
```diff
@@ -52,6 +54,20 @@ public void testBuild() {
...
+      //tag::basic-example[]
+      EmbeddedCacheManager manager = new DefaultCacheManager("configs/my-custom-config.xml");
+      Cache defaultCache = manager.getCache();
+      //end::basic-example[]
+      //tag::named-cache-retrieval[]
+      Cache someNamedCache = manager.getCache("local");
+      //end::named-cache-retrieval[]
+
+      EmbeddedCacheManager manager2 = new DefaultCacheManager("configs/my-custom-config2.xml");
+      EmbeddedCacheManager manager3 = new DefaultCacheManager("configs/my-custom-config3.xml");
```
Comment (galderz) [inline core/src/test/java/org/infinispan/configuration/ConfigurationUnitTest.java]: Seems to me you might be leaking cache managers here? They are not being shutdown...

I know is for documentation, but in the past I've had issues with creating cache managers without using the `TestCacheManagerFactory` methods. These issues were the result of the cache managers not being isolated in our concurrent test suite. 

Not sure if you'd have problems too, but I'd favour such examples to be somewhere else where maybe tests that verify code examples can run sequentially, where you don't run such risks.

File: core/src/test/resources/configs/my-custom-config.xml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (galderz) [inline core/src/test/resources/configs/my-custom-config.xml]: Happy with configs but I wonder if the name of the files could be something more meaningful. As a reader, you might wonder what might be the difference between `my-custom-config23.xml` and `my-custom-config534.xml` ;)

**Comment by rvansa** [comment]
@Holmistr This is rather dormant, will you respond to Galder?

**Comment by tristantarrant** [comment]
I guess this PR is now orphaned and will need someone else's love.

**Comment by tristantarrant** [comment]
Closing until someone picks it up again

**Comment by rvansa** [comment]
@oraNod I think you are looking for this PR.

**Pull request closed without merging**